src/zip.ts

```diff
@@ -151,3 +156,31 @@ export function zip64ExtraField(file: ZipFileDescription, offset: bigint) {
   header.setBigUint64(20, offset, true)
   return makeUint8Array(header)
 }
+
+export function unicodePathExtraField(file: ZipFileDescription) {
+  if (!file.utf8) {
+    // make typescript compiler happy
+    throw new Error('file.utf8 must be set')
+  }
+
+  // Value         Size        Description
+  // -----         ----        -----------
+  // 0x7075        Short       tag for this extra block type ("up")
+  // TSize         Short       total data size for this block
+  // Version       1 byte      version of this extra field, currently 1
+  // NameCRC32     4 bytes     File Name Field CRC32 Checksum
+  // UnicodeName   Variable    UTF-8 version of the entry File Name
+
+  const header = makeBuffer(unicodeHeaderLength)
+  header.setUint16(0, 0x7075, true)
+  header.setUint16(2, unicodeHeaderLength - 4 + file.utf8.byteLength, true)
+  header.setUint8(4, 1)
+  header.setUint32(5, crc32(file.encodedName), true)
+
+  // concat the UnicodeName
+  const headerA = makeUint8Array(header)
+  const field = new Uint8Array(headerA.byteLength + file.utf8.byteLength)
+  field.set(new Uint8Array(headerA), 0);
+  field.set(new Uint8Array(file.utf8), headerA.byteLength)
```

Comment:
Instead of concatenating those buffers, you can just return the fixed-length part here, and push the actual filename separately (since you already have a buffer for it). That way, no unnecessary copying.